src/main/scala/strava/FetchActivities.scala

```diff
@@ -30,43 +29,42 @@ object FetchActivities extends IOApp with Http4sClientDsl[IO] {
       val clientId = ClientId(args.head)
       val clientSecret = ClientSecret(args.tail.head)
       val yearToFetch = Year.of(args.tail.tail.head.toInt)
-      Blocker[IO].use { blocker =>
-        BlazeClientBuilder[IO](global).resource.use { client =>
-          getBearerToken(blocker, client, clientId, clientSecret).flatMap { bearerToken =>
-            fetchActivitiesForYearJson(client, bearerToken, yearToFetch)
-              .compile.toVector
-              .flatTap(activities => IO(println(s"Fetched ${activities.size} activities")))
-              .flatMap(ActivityStorage.writeActivitiesJson(blocker, _))
-          }
-        }
+      val client = JavaNetClientBuilder[IO].create
+      getBearerToken(client, clientId, clientSecret).flatMap { bearerToken =>
+        fetchActivitiesForYearJson(client, bearerToken, yearToFetch)
+          .compile.toVector
+          .flatTap(activities => IO(println(s"Fetched ${activities.size} activities")))
+          .flatMap(ActivityStorage.writeActivitiesJson)
       }.as(ExitCode.Success)
     }
   }
 
-  def getBearerToken(blocker: Blocker, client: Client[IO], clientId: ClientId, clientSecret: ClientSecret): IO[BearerToken] = {
-    getAuthorizationCode(blocker, clientId).flatMap(fetchBearerToken(client, clientId, clientSecret, _))
+  def getBearerToken(client: Client[IO], clientId: ClientId, clientSecret: ClientSecret): IO[BearerToken] = {
+    getAuthorizationCode(clientId).flatMap(fetchBearerToken(client, clientId, clientSecret, _))
   }
 
-  def getAuthorizationCode(blocker: Blocker, clientId: ClientId): IO[AuthorizationCode] = {
+  def getAuthorizationCode(clientId: ClientId): IO[AuthorizationCode] = {
     Deferred[IO, AuthorizationCode].flatMap { deferredAuthCode =>
-      BlazeServerBuilder[IO](global)
-        .bindHttp(port = 0)
-        .withHttpApp { 
-          object CodeParam extends QueryParamDecoderMatcher[String]("code")
-          HttpRoutes.of[IO] {
-            case GET -> Root / "exchange_token" :? CodeParam(code) =>
-              deferredAuthCode.complete(AuthorizationCode(code)).as(Response(Status.Ok))
-          }.orNotFound
-        }
-        .stream.flatMap { server =>
-          val port = server.address.getPort
-          Stream.eval(requestAuthCode(blocker, clientId, port) *> deferredAuthCode.get)
-        }.compile.lastOrError
+      Dispatcher[IO].use { dispatcher =>
+        BlazeServerBuilder[IO](global, dispatcher)
```

Comment:
Note: creating a Dispatcher is super lightweight. I think `BlazeServerBuilder` should do this internally and not require the client to do so.

Reply:
Is there a reason to customize it or share it?  I've seen a lot of `withDispatcher` blowing around the CE3 branch.

Reply:
Nope, only argument I've seen is mocking it out in tests, but I think that's asking for double.